tox.ini

```
@@ -108,16 +98,15 @@ commands =
     # Run pex and ensure it doesn't fail
     python -Werror {toxinidir}/px.pex
     # Test version string vs git
-    /bin/bash -x -c 'test "$("{toxinidir}/px.pex" --version)" = "$(git describe --dirty)"'
+    /bin/bash -x -c '[[ "$("{toxinidir}/px.pex" --version)" =~ ^"$(git describe | cut -f1,2 -d.)" ]]'
```

Would cutting by `-` be a better version number comparison?

```suggestion
    /bin/bash -x -c '[[ "$("{toxinidir}/px.pex" --version)" =~ ^"$(git describe | cut -f1 -d-)" ]]'
```

---

Actually it would, let's just add this.

---

Well that didn't work, reverted.

@branchvincent why does CI fail if all three parts of the version number are compared?

[Here](https://github.com/walles/px/actions/runs/17257123354/job/48971214035?pr=138#step:6:1082)'s what CI said after my change:
```
test-package: commands[3]> /bin/bash -x -c '[[ "$("/Users/runner/work/px/px/px.pex" --version)" =~ ^"$(git describe | cut -f1 -d-)" ]]'
++ git describe
++ cut -f1 -d-
+ [[ 3.6.14.dev4+g73abbfae4 =~ ^3\.6\.13 ]]
test-package: exit 1 (0.11 seconds) /Users/runner/work/px/px> /bin/bash -x -c '[[ "$("/Users/runner/work/px/px/px.pex" --version)" =~ ^"$(git describe | cut -f1 -d-)" ]]' pid=1733
```

---

Answered here, nvm: https://stackoverflow.com/a/74404703/473672

---

Ah yea sorry I should have called that out. See also https://stackoverflow.com/a/56973100 for the why (which is to adhere to PEP 440's versioning scheme, and so that all dev versions after a git tag compare as newer)